ocular/src/chain/client.rs

```diff
@@ -24,20 +26,28 @@ pub struct ChainClient {
 }
 
 impl ChainClient {
-    pub fn new(chain_name: &str) -> Result<Self, ChainClientError> {
-        let chain = executor::block_on(async { get_chain(chain_name).await })?;
-        let config = chain.get_chain_config()?;
-        let rpc_client = new_rpc_client(config.rpc_address.as_str())?;
-        let keyring = Keyring::new_file_store(None).expect("Could not create keyring.");
-
-        Ok(ChainClient {
-            config,
-            keyring,
-            rpc_client,
-        })
+    pub fn new(chain: Chains) -> Result<Self, ChainClientError> {
+        get_client(chain.as_str())
+    }
+
+    pub fn new_unsupported(chain_name: &str) -> Result<Self, ChainClientError> {
+        get_client(chain_name)
```

Comment:
what if a user incorrectly tries to get an unsupported chain that's already supported? E.g. I put in new_unsupported = "evmos"; should we automatically map that to a Chains obj?

Reply:
The true reference name is a `&str` so this will still work. Notice they both call the same `get_client` method that takes a `&str`

Reply:
Ah yup see that now. Maybe instead of unsupported we say custom? Just to not indicate it's entirely unsupported by the client?

Reply:
I think custom implies more meaning than we want. Maybe `new_with_str` or `try_new` (since it returns a `Result`)?

Reply:
hmm yeah thats a good point. Maybe new_chain_name?

Reply:
That sounds like we're creating a chain name to me. Maybe we leave it for now and PR later if we come up with a better name?